mcs/class/System/System.Net/ResponseStream.cs

```diff
@@ -81,19 +81,25 @@ public override void Close ()
 				byte [] bytes = null;
 				MemoryStream ms = GetHeaders (true);
 				bool chunked = response.SendChunked;
-				if (ms != null) {
-					long start = ms.Position;
-					if (chunked && !trailer_sent) {
-						bytes = GetChunkSizeBytes (0, true);
-						ms.Position = ms.Length;
-						ms.Write (bytes, 0, bytes.Length);
+				if (stream.CanWrite) {
+					try {
+						if (ms != null) {
+							long start = ms.Position;
+							if (chunked && !trailer_sent) {
+								bytes = GetChunkSizeBytes (0, true);
+								ms.Position = ms.Length;
+								ms.Write (bytes, 0, bytes.Length);
+							}
+							InternalWrite (ms.GetBuffer (), (int) start, (int) (ms.Length - start));
+							trailer_sent = true;
+						} else if (chunked && !trailer_sent) {
+							bytes = GetChunkSizeBytes (0, true);
+							InternalWrite (bytes, 0, bytes.Length);
+							trailer_sent = true;
+						}
+					} catch (IOException ex) {
```

Comment:
From our git history is hard to tell why we are trying to fulfill responses before closing HttpListener.
From reference sources it is not possible to find any code doing that.
http://referencesource.microsoft.com/#System/net/System/Net/HttpListener.cs,1048
It could be implemented on the native calls, those may return errors but they are only logged, not thrown. 
exs:
 - http://referencesource.microsoft.com/#System/net/System/Net/HttpListener.cs,953
 - http://referencesource.microsoft.com/#System/net/System/Net/HttpListener.cs,907

From a logic point of view this code is trying to send data to client with the intent of not letting them with pending request, but if they already closed the connection I don't think we should fail by not being able of informing peers of something they don't need anymore.

Not sure if we throw other IOException, but as reference source doesn't seem to be able to cause an exception executing the same logic I don't think we should.